lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -81,10 +103,65 @@ def test_sort_order(self):
         response = self.api_response()
         for course_num in range(num_courses):
             self.assertEqual(
-                response.data[course_num]['course']['id'],  # pylint: disable=no-member
+                response.data[course_num]['course']['id'],
                 unicode(courses[num_courses - course_num - 1].id)
             )
 
+    @patch.dict(
+        settings.FEATURES, {'ENABLE_PREREQUISITE_COURSES': True, 'MILESTONES_APP': True, 'DISABLE_START_DATES': False}
+    )
+    def test_courseware_access(self):
+        seed_milestone_relationship_types()
+        self.login()
+
+        course_with_prereq = CourseFactory.create(start=self.YESTERDAY, mobile_available=True)
+        prerequisite_course = CourseFactory.create()
+        set_prerequisite_courses(course_with_prereq.id, [unicode(prerequisite_course.id)])
+
+        # Create list of courses with various expected courseware_access responses and corresponding expected codes
+        courses = [course_with_prereq,
+                   CourseFactory.create(start=self.TOMORROW, mobile_available=True),
+                   CourseFactory.create(visible_to_staff_only=True, mobile_available=True),
+                   CourseFactory.create(start=self.YESTERDAY, mobile_available=True, visible_to_staff_only=False)]
+
+        expected_error_codes = [MilestoneError().error_code,  # 'unfulfilled_milestones'
+                                StartDateError(self.TOMORROW).error_code,  # 'course_not_started'
+                                VisibilityError().error_code,  # 'not_visible_to_user'
+                                None]
+
+        # Enroll in all the courses
+        for course in courses:
+            self.enroll(course.id)
+
+        # Verify courses have the correct response through error code. Last enrolled course is first course in response
+        response = self.api_response()
+        for course_num in range(len(courses)):
```

Comment:
I take that back. It looks like `course_num` is used in other places in this file, so consistency wins.